File: chronoFlow-ui/src/utils/status.ts
import type { TagProps } from 'ant-design-vue'

export function statusColor(status: string): TagProps['color'] {
  const map: Record<string, TagProps['color']> = {
    online: 'success',
    offline: 'error',
    running: 'processing',
    killing: 'warning',
    stopped: 'default',
    success: 'success',
    failed: 'error',
    killed: 'warning',
    skipped: 'default',
  }
  return map[status] || 'default'
}

export function statusText(status: string): string {
  const map: Record<string, string> = {
    online: '在线',
    offline: '离线',
    running: '运行中',
    killing: '终止中',
    stopped: '已停止',
    success: '成功',
    failed: '失败',
    killed: '已终止',
    skipped: '已跳过',
    manual: '手动',
    cron: '定时',
  }
  return map[status] || status || '-'
}

export function isActiveLogStatus(status: string): boolean {
  return status === 'running' || status === 'killing'
}

File: chronoFlow-admin/internal/biz/status.go
package biz

const (
	ScheduleStatusStopped = "stopped"
	ScheduleStatusRunning = "running"

	ExecutorStatusOnline  = "online"
	ExecutorStatusOffline = "offline"

	TriggerTypeManual = "manual"
	TriggerTypeCron   = "cron"

	JobLogStatusRunning = "running"
	JobLogStatusKilling = "killing"
	JobLogStatusSuccess = "success"
	JobLogStatusFailed  = "failed"
	JobLogStatusTimeout = "timeout"
	JobLogStatusSkipped = "skipped"
	JobLogStatusKilled  = "killed"

	AlertStatusNone    = "none"
	AlertStatusPending = "pending"
	AlertStatusSent    = "sent"
	AlertStatusFailed  = "failed"
	AlertStatusSkipped = "skipped"
)

func IsFinalJobLogStatus(status string) bool {
	switch status {
	case JobLogStatusSuccess, JobLogStatusFailed, JobLogStatusTimeout, JobLogStatusSkipped, JobLogStatusKilled:
		return true
	default:
		return false
	}
}

func CanCallbackUpdateJobLogStatus(status string) bool {
	return status == JobLogStatusRunning || status == JobLogStatusKilling
}
